measurement/process/meta.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+import logging
+import os
+
+from measurement.files import fileutils
+
+
+def find_process_by_port(port=None):
+    process_list = []
+    if not port:
+        logging.fatal("No process listening port specified.")
+        return
+
+    # literate over all file descriptors and build a socket address -> pid map
+    def build_inode_to_pid_map():
+        result = {}
+        for entry in os.scandir("/proc"):
+            # find all PIDs
+            if str.isdigit(entry.name):
+                try:
+                    for _fd in os.scandir("/proc/%s/fd" % entry.name):
+                        _fd_target = os.readlink(_fd.path)
+                        if not str.startswith(_fd_target, "socket"):
+                            continue
+                        _socket = _fd_target.split(":[")[-1][:-1]
```

Comment:
If it's a broken socket, eg `3: broken symbolic link to socket:[418388778]` , what should we do?

Reply:
It's trying to find the PID that holds a given listening port, so if the socket is broken, we're safe to ignore here.